.github/workflows/test.yml

```diff
@@ -2,33 +2,23 @@ name: Tests
 
 on:
   push:
-    branches: [main]
   pull_request:
-    branches: [main]
+  workflow_dispatch:
+  schedule: [ cron: "42 0 7 * *" ] # https://crontab.guru/#42_0_7_*_*
   workflow_run:
-    workflows: ["Update"]
-    types:
-      - completed
+    workflows: [ "Update" ]
+    types: [ completed ]
 
 jobs:
   test:
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-        ruby-version:
-          - "3.1"
-          - "3.2"
-          - "3.3"
-          - "3.4"
-
-    runs-on: ${{ matrix.os }}
-
+    runs-on: ubuntu-latest
+    strategy: {matrix: {ruby: [ '3.0', 3.1, 3.2, 3.3, 3.4 ]}}
     steps:
-      - uses: actions/checkout@v4
-      - name: Set up Ruby ${{ matrix.ruby-version }}
-        uses: ruby/setup-ruby@v1
+      - uses: step-security/harden-runner@6c439dc8bdf85cadbbce9ed30d1c7b959517bc49 # v2.12.2
+        with: { egress-policy: audit }
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
FYI: pinning to shas is the recommended approach for security (because branches and tags are mutable and could be compromised via force pushes).

the trailing comment with version number _IS_ bumped by dependabot so it will stay accurate.

Reply:
(there is some irony, however, that even the shas are not hack-proof. to my knowledge, if someone were to push a ref that collides with a sha, github would use the ref, not the sha. 😱 )